crates/hir-ty/src/lower.rs

```diff
@@ -509,7 +509,14 @@ impl<'a> TyLoweringContext<'a> {
                         TyKind::Placeholder(to_placeholder_idx(self.db, param_id.into()))
                     }
                     ParamLoweringMode::Variable => {
-                        let idx = generics.param_idx(param_id.into()).expect("matching generics");
+                        let idx = match generics.param_idx(param_id.into()) {
+                            // FIXME: never crash if this involves const generic associated types
+                            None if generics.is_gat => {
```

Comment:
let's just do a `never!()` check here instead of checking `is_gat`

Reply:
This will make the test fail though.

Reply:
Yeah, I would be fine with that. It's a bit unfortunate that we can't test that it doesn't crash, but I'd rather have the crashing test as a reminder that it's still broken.